Error on E701 Multiple statements on one line (colon)

<!--
Thank you for taking the time to report an issue! We're glad to have you involved with Ruff.

If you're filing a bug report, please consider including the following information:

* A minimal code snippet that reproduces the bug.
* The command you invoked (e.g., `ruff /path/to/file.py --fix`), ideally including the `--isolated` flag.
* The current Ruff settings (any relevant sections from your `pyproject.toml`).
* The current Ruff version (`ruff --version`).
-->
Ruff v0.0.245 raises a `Multiple statements on one line (colon)` error while v0.0.244 did not.
`total_color = success_color_fn if all_green else lambda x: x`
`openlibrary/i18n/__init__.py:234:70: E701 Multiple statements on one line (colon)`

--- a/crates/ruff/resources/test/fixtures/pycodestyle/E70.py
+++ b/crates/ruff/resources/test/fixtures/pycodestyle/E70.py
@@ -44,3 +44,5 @@ a: List[str] = []
 #:
 if a := 1:
     pass
+#:
+func = lambda x: x** 2 if cond else lambda x:x

--- a/crates/ruff/src/rules/pycodestyle/rules/compound_statements.rs
+++ b/crates/ruff/src/rules/pycodestyle/rules/compound_statements.rs
@@ -182,6 +182,21 @@ pub fn compound_statements(lxr: &[LexResult]) -> Vec<Diagnostic> {
         }
 
         match tok {
+            Tok::Lambda => {
+                // Reset.
+                def = None;
+                colon = None;
+                class = None;
+                elif = None;
+                else_ = None;
+                except = None;
+                finally = None;
+                for_ = None;
+                if_ = None;
+                try_ = None;
+                while_ = None;
+                with = None;
+            }
             Tok::If => {
                 if_ = Some((start, end));
             }

